zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/ReschedulingTaskTest.java

```diff
@@ -40,6 +45,71 @@ void shouldRescheduleTaskOnError() {
         .schedule(task, 10L, TimeUnit.MILLISECONDS);
   }
 
+  @Test
+  void shouldLogEvery10ThReoccurringError() {
+    // given
+    final Logger loggerMock = Mockito.spy(Logger.class);
+    final Exception exception = new RuntimeException("failure");
+    final var task =
+        new ReschedulingTask(
+            new FailingJob().setException(exception), 1, 10, 10, executor, loggerMock);
+
+    // when
+    task.run();
+
+    // then - every 10th error is logged on error level, others on debug
+    final String errorMessage =
+        "Error occurred while performing a background task FailingJob; error message `failure`; operation will be retried";
+    final ArgumentCaptor<Exception> exceptionCaptor = ArgumentCaptor.forClass(Exception.class);
+    final var inOrder = Mockito.inOrder(loggerMock);
+    inOrder
+        .verify(loggerMock, Mockito.timeout(5_000).times(1))
+        .error(eq(errorMessage), exceptionCaptor.capture());
+    assertThat(exceptionCaptor.getValue().getCause()).isEqualTo(exception);
+    for (int i = 1; i <= 9; i++) {
+      inOrder.verify(loggerMock, Mockito.timeout(5_000).times(1)).debug(errorMessage);
+    }
+    inOrder
+        .verify(loggerMock, Mockito.timeout(5_000).times(1))
+        .error(eq(errorMessage), exceptionCaptor.capture());
+    assertThat(exceptionCaptor.getValue().getCause()).isEqualTo(exception);
+    for (int i = 1; i <= 9; i++) {
+      inOrder.verify(loggerMock, Mockito.timeout(5_000).times(1)).debug(errorMessage);
+    }
+  }
+
+  @Test
+  void shouldLogEveryNewError() {
+    // given
+    final Logger loggerMock = Mockito.spy(Logger.class);
+    final Exception exception = new RuntimeException("failure");
+    final var task =
+        new ReschedulingTask(
+            new FailingJob().setException(exception).setRandomExceptionMessage(true),
+            1,
+            10,
+            10,
+            executor,
+            loggerMock);
+
+    // when
+    task.run();
+
+    // then - when errors are different they are all logged on error level
+    final String errorMessageSubstring =
```

Comment:
:wrench: Check out the `GrpcErrorMapperTest` for an example of how to unit test logging without mocking. You can use a recording appender, such that the logger ends up logging to it and not to the STDOUT or whatever, letting you run assertions on the log messages.

Of course, this implies having the class accept different loggers, but the `ReschedulingTask` already does :)